.github/workflows/redis-rs.yml

```diff
@@ -0,0 +1,124 @@
+              working-directory: ./glide-core/redis-rs/redis/src
+
+            - name: Lint redis-rs
+              shell: bash
+              run: |
+                  cargo fmt --all -- --check
+                  cargo clippy -- -D warnings
+                  cargo install --locked cargo-deny
+                  cargo deny check all --config ${GITHUB_WORKSPACE}/deny.toml --exclude-dev all
+              working-directory: ./glide-core/redis-rs/redis
+
+            - name: Test
+              run: |
+                  cargo test --release -- --test-threads=1 | tee ../test-results.xml
+                  echo "### Tests passed :v:" >> $GITHUB_STEP_SUMMARY
+              working-directory: ./glide-core/redis-rs/redis/src
+
+            - name: Upload test reports
+              if: always()
+              continue-on-error: true
+              uses: actions/upload-artifact@v4
+              with:
+                  name: test-reports-redis-rs
+                  path: ./glide-core/redis-rs/redis/test-results.xml
+
+            - name: Run benchmarks
+              run: |
+                  cargo bench | tee bench-results.xml
+              working-directory: ./glide-core/redis-rs/redis
+
+            - name: Upload benchmark results
+              if: always()
+              continue-on-error: true
+              uses: actions/upload-artifact@v4
+              with:
+                  name: benchmark-results-redis-rs
+                  path: ./glide-core/redis-rs/redis/bench-results.xml
+
+            - name: Test docs
+              run: |
+                  cargo test --doc
+              working-directory: ./glide-core/redis-rs/redis/src
+
+            - name: Security audit
+              run: |
+                  cargo audit | tee audit-results.txt
+                  if grep -q "Crate: " audit-results.txt; then
+                    echo "## Security audit results summary: Security vulnerabilities found :exclamation: :exclamation:" >> $GITHUB_STEP_SUMMARY
+                    echo "Security audit results summary: Security vulnerabilities found"
+                  else
+                    echo "### Security audit results summary: All good, no security vulnerabilities found :closed_lock_with_key:" >> $GITHUB_STEP_SUMMARY
+                    echo "Security audit results summary: All good, no security vulnerabilities found"
+                  fi
+              working-directory: ./glide-core/redis-rs/redis
+
+            - name: Run cargo machete
+              run: |
+                  cargo install cargo-machete
+                  cargo machete | tee machete-results.txt
+                  if grep -A1 "cargo-machete found the following unused dependencies in this directory:" machete-results.txt | sed -n '2p' | grep -v "^if" > /dev/null; then
+                    echo "Machete results summary: Unused dependencies found" >> $GITHUB_STEP_SUMMARY
+                    echo "Machete results summary: Unused dependencies found"
```

Comment:
You can upload report as an artifact

Reply:
Somebody check it? Here i just want to pop a noticeable message, it is not an error, Im absolutely fine with adding report also, the question is if someone do something with it.

Reply:
We will if CI is red.
BTW is `machete` returns non-zero exit code on error? if yes, GHA aborts the script and nothing come to the summary. Probably you need to suppress exit code from `machete` and rethrow an error (`exit 1`) or add one more step to read the output (with `if: always()`).

Reply:
Its not, but i will fail the CI manually